bash2048.sh

```diff
@@ -262,6 +264,70 @@ function reload_game {
   index_max=board_size-1
 }
 
+# print the content of the highscores file
+function show_highscores {
+    output="\nHighscore Name Date\n\n"
+    while read line; do
+        output="${output}${line}\n"
+    done < "${highscore_file}"
+    printf "${output}" | column -t
+}
+
+# check if a score is a high score (in top 10 highest scores)
+# input:
+#         $1: score value
+# output:
+#         0 if true
+#         1 if false
+function is_high_score(){
+    least_score=$(cut -f 1| sort -rn| tail -n 1)
+    if [[ "${1}" -ge "${least_score}" ]]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# save a new highscore
+# inputs:
+#         $1: score value
+#         $2: score timestamp
+function save_highscore {
+    # highscore file doesn't exist or is not readable
+    if [[ ! -f "${highscore_file}" ]] || [[ ! -r "${highscore_file}" ]]; then
+        touch "${highscore_file}"
+    fi
+    echo ""
+    read -p "Please enter your name: " name
+    new_highscore="${1} ${name} $(date -d @${2})"
+    number_of_saved_highscores=$(wc -l "${highscore_file}")
+    if [[ "${number_of_saved_highscores}" = "${highscore_count_max}" ]]; then
+        # replace the last line (lowest highscore) if the limit is reached
+        sed -i "$s/.*/${new_highscore}/" "${highscore_file}"
+    else 
+        # append otherwise
+        echo "${new_highscore}" >> "${highscore_file}"
+    fi
+    # reverse sort (greatest to lowest) the highscores file
+    sort -k1 -nr "${highscore_file}"
```

Comment:
if we sort the file here, do we need to sort it again in the is_high_score function?

Reply:
My understanding (I wrote this code 3 years ago) is that the lines stored in high scores file are not ordered. Instead, they're appended to the end of the file by « order of arrival ». The line you're pointing is used to display the highscores file after saving a new score (I guess it's optional.) The other sort is used to compare a new score with the least high score before eventually adding it to the list.

Reply:
ok, that seems reasonable